src/main/java/core/basesyntax/impl/WritingToNewFileImpl.java

```diff
@@ -0,0 +1,32 @@
+package core.basesyntax.impl;
+
+import core.basesyntax.service.WritingToNewFile;
+import core.basesyntax.db.FruitShopDao;
+import core.basesyntax.service.CantWorkWithThisFileException;
+
+import java.io.BufferedWriter;
+import java.io.FileWriter;
+import java.io.IOException;
+
+public class WritingToNewFileImpl implements WritingToNewFile {
+    private FruitShopDao fruitShopDao;
+
+    public WritingToNewFileImpl(FruitShopDao fruitShopDao) {
+        this.fruitShopDao = fruitShopDao;
+    }
+
+    @Override
+    public void writing(String fileName) {
```

Comment:
The method name `writing` is not descriptive enough and does not follow Java method naming conventions. Consider renaming it to something more indicative of its functionality, such as `writeReportToFile`.